### PR TITLE
RaftMerge: implement a simplified raft model.

### PR DESCRIPTION
**Pull request by foreverbell**
This is the first part of my plan to implement raft merge protocol in TLA+. This part comprises a simplified raft model with these properties.

* No leader election and heartbeat, the leader is fixed.
* No network partition. All messages will be eventually delivered (may out of order).
* Relax the concept of quorum (motivation explained in comment).

The body part of Raft merge will be added in follow-up PRs.

@GregoryIan @siddontang

### CODE REVIEW

**Comment by IANTHEREAL** [comment]
LGTM

File: RaftMerge/RaftMerge.tla [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  [ is_leader    : BOOLEAN,
+    logs         : Seq(LogType),
+    commit_index : Nat,
+    apply_index  : Nat,
+    match_index  : [Store -> Nat]  \* Only available on leader.
+                                   \* Initialized to zeroes on followers.
+  ]
+
+TypeInvariant ==
+  /\ raft   \in [Store -> [Region -> RaftType]]
```
Comment (siddontang) [inline RaftMerge/RaftMerge.tla]: remove extra spaces

Comment (foreverbell) [reply RaftMerge/RaftMerge.tla]: Done.

File: RaftMerge/RaftMerge.tla [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    /\ LogAppliable(i, r)
+    /\ raft' = [raft EXCEPT ![i][r].apply_index = next_index]
+    /\ UNCHANGED <<messages, region, client_vars>>
+
+-------------------------------------------------------------------------------
+
+\* Specification of Raft merge.
+Next ==
+  \* Raft actions.
+  \/ \E i,j \in Store : \E r \in Region : AppendEntries(i, j, r)
```
Comment (siddontang) [inline RaftMerge/RaftMerge.tla]: `i, j`

File: RaftMerge/RaftMerge.tla [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+                              apply_index  |-> 0,
+                              match_index  |-> [j \in Store |-> 0]
+                             ]
+                           ]
+                         ]
+     IN
+       raft = MarkLeader(MarkLeader(no_leader_raft, LeaderA, RegionA),
+                         LeaderB,
+                         RegionB)
+  /\ region = TRUE
```
Comment (siddontang) [inline RaftMerge/RaftMerge.tla:226]: why TRUE here? where to use region?

Comment (foreverbell) [reply RaftMerge/RaftMerge.tla:226]: This is a variable will be used in later PR.

**Review by siddontang — APPROVED** [review]
LGTM

**Pull request merged**